src/theme.cpp

```diff
@@ -463,7 +463,7 @@ theme::status_item::status_item(std::size_t sw, std::size_t sh, const config& cf
 	}
 
 	if(cfg.has_attribute("font_rgb")) {
-		font_rgb_ = color_t::from_rgb_string(cfg["font_rgb"]);
+		font_rgb_ = color_t::from_rgb_string(cfg["font_rgb"].str_view());
```

Comment:
I think this is actually _very_ terrible. Consider what would happen if you _forget_ to call `str_view()` here. If I'm not mistaken, that'll result in the config_attribute_value copying its value into a string, which is then implicitly converted to a `string_view` in order to be passed to `from_rgb_string`.

So… what is the lifetime of that temporary string? I don't recall the exact rules off the top of my head. Will it still be valid while `from_rgb_string` is running?

But, even if there isn't a dangling reference issue here, it's bad that the default behaviour is to copy and you need extra effort to get reference behaviour.

Reply:
Without it, it gives a compiler error, not the erroneous view to a tangling temporary.

Reply:
> So… what is the lifetime of that temporary string? I don't recall the exact rules off the top of my head. Will it still be valid while from_rgb_string is running?

c++ keeps such temporaries objects alive until the function finished exceuting, i'm pretty sure thqt we use things like `function_that_takes_str_view(function_that_returns_a_string_by_value())` all the time

Reply:
> Without it, it gives a compiler error, not the erroneous view to a tangling temporary.

Is it an ambiguous conversion error, or what?

Reply:
Not home right now so I can’t double check, but CAR is implicitly convertible to string, and not string_view.

Reply:
The exact error is
```
'color_t color_t::from_rgb_string(std::string_view)': cannot convert argument 1 from 'const config::attribute_value' to 'std::string_view'
```

Reply:
Oh, I guess this is because an implicit conversion can use at most one _user-defined_ conversion, or something like that. If it would require multiple user-defined conversions, then it's an error.

So that definitely is far less bad than I thought if it's a compile-time error. However, the last sentence of my original post still applies. I think it would be more logical to make the `string_view` conversion implicit in CAR, while a `string` conversion must be explicit. That isn't possible with the implementation you have here, though, since your `string_view` conversion fails for anything that's not stored as a string. But see my post above this thread for other ideas.